src/test/java/org/jboss/eap/qa/jarest/DistributionTestCase.java

```diff
@@ -106,4 +108,66 @@ void atMostMRv9JARsPresent() throws IOException {
+                                    jar.getAutomaticModuleNameFromManifest().isPresent())
+                                    .as("Artifact %s has no Automatic-Module-Name", jar.baseFileName())
+                                    .isTrue();
+                        }
+                );
+        softly.assertAll();
+    }
+
+    @Test
+    void checkExpectedAutomaticModuleNames() throws IOException {
+        SoftAssertions softly = new SoftAssertions();
+        Map<String, String> expectedNames = config.expectedAutomaticModuleNames();
+        distributionDir.jars()
+                .filter(jar -> jar.getAutomaticModuleNameFromManifest().isPresent())
+                .filter(jar -> expectedNames.containsKey(jar.baseFileNameWithoutVersion()))
+                .sorted(Comparator.comparing(jar -> jar.baseFileName()))
+                .forEach(jar -> {
+                            String moduleName = jar.getAutomaticModuleNameFromManifest().get();
+//                            System.out.println(jar
+//                                    + "\t-\t" + jar.baseFileNameWithoutVersion()
+//                                    + "\t-\t" + moduleName);
+                            softly.assertThat(
+                                    expectedNames.get(jar.baseFileNameWithoutVersion()).equals(moduleName))
+                                    .as("Artifact %s has unexpected Automatic-Module-Name %s, expected was %s",
+                                            jar.baseFileName(), moduleName, expectedNames.get(jar.baseFileNameWithoutVersion()))
+                                    .isTrue();
+                        }
+                );
+        softly.assertAll();
+    }
+
+    @Test
+    void listUntrackedFilesWithAutomaticModuleNames() throws IOException {
+        SoftAssertions softly = new SoftAssertions();
+        Map<String, String> expectedNames = config.expectedAutomaticModuleNames();
+        distributionDir.jars()
+                .filter(jar -> jar.getAutomaticModuleNameFromManifest().isPresent())
+                .filter(jar -> ! expectedNames.containsKey(jar.baseFileNameWithoutVersion()))
+                .sorted(Comparator.comparing(jar -> jar.baseFileName()))
+                .forEach(jar -> {
+                            String moduleName = jar.getAutomaticModuleNameFromManifest().get();
+                            System.out.println(jar
+                                    + "\t-\t" + moduleName);
+//                            softly.assertThat(
+//                                    expectedNames.get(jar.baseFileNameWithoutVersion()).equals(moduleName))
+//                                    .as("Artifact %s has unexpected Automatic-Module-Name %s, expected was %s",
+//                                            jar.baseFileName(), moduleName, expectedNames.get(jar.baseFileNameWithoutVersion()))
+//                                    .isTrue();
+                        }
+                );
+        softly.assertAll();
```

Comment:
I cannot see any soft assertion in this method - I suppose this will be a subject to change when the criteria are cleared up with devs?

Reply:
This is to be done, this test is used atm just to list those files and adjust tracking spreadsheet.
Output is used as base for discussion with devs.